semantic_release/cli/commands/changelog.py

```diff
@@ -62,10 +52,55 @@ def changelog(ctx: click.Context, release_tag: str | None = None) -> None:
             )
             ctx.exit(0)
 
-        changelog_text = render_default_changelog_file(env)
-        with open(str(changelog_file), "w+", encoding="utf-8") as f:
-            f.write(changelog_text)
+        with open(str(changelog_file), "a+", encoding="utf-8") as f:
+            f.seek(0)
+            existed_changelog_text = f.read()
+            f.truncate()
+            existed_changelog_sections = existed_changelog_text.split(CHANGELOG_PLACEHOLDER)
+            if len(existed_changelog_sections) > 1:
+                log.warning("Deprecation: Placeholder is detected, only new changes will be added.")
+                rh = ReleaseHistory.from_git_history(
```

Comment:
We always iterate in reverse order over the commits, which means the latest release is the first one inserted into `ReleaseHistory.released`.

You can get the last release by doing:

```python3
full_rh = ReleaseHistory.from_git_history(<the existing params>)
latest_release_version = next(iter(full_rh.unreleased), None)
latest_release = full_rh.released.get(latest_release_version, {})

rh = ReleaseHistory(unreleased=full_rh.unreleased, released=latest_release)
```

This works because `dict`s remember their insertion order so `next` will find the oldest-inserted key

Reply:
with these suggestions I'm not sure how to keep the existing changelog, as I don't want to override previous changelogs, only add the latest changelog